packages/edit-site/src/components/global-styles/hooks.js

```diff
@@ -0,0 +1,210 @@
+	];
+};
+
+export function useSetting( path, blockName, source = 'all' ) {
+	const [ baseConfig, userConfig, setUserConfig ] = useGlobalStylesConfig();
+	const finalPath = ! blockName
+		? `settings.${ path }`
+		: `settings.blocks.${ blockName }.${ path }`;
+
+	const getBaseSetting = () => {
+		const result = get( baseConfig, finalPath );
+		if ( PATHS_WITH_MERGE[ path ] ) {
+			return result.theme ?? result.core;
+		}
+	};
+
+	const setSetting = ( newValue ) => {
+		const newUserConfig = cloneDeep( userConfig );
+		set( newUserConfig, finalPath, newValue );
+		setUserConfig( newUserConfig );
+	};
+
+	let result;
+	switch ( source ) {
+		case 'all':
+			result = get( userConfig, finalPath ) ?? getBaseSetting();
+			break;
+		case 'user':
+			result = get( userConfig, finalPath );
+			break;
+		case 'base':
+			result = getBaseSetting();
+			break;
+		default:
+			throw 'Unsupported source';
+	}
+
+	return [ result, setSetting ];
+}
+
+export function useStyle( path, blockName, source = 'all' ) {
+	const [ baseConfig, userConfig, setUserConfig ] = useGlobalStylesConfig();
+	const finalPath = ! blockName
+		? `styles.${ path }`
+		: `styles.blocks.${ blockName }.${ path }`;
+
+	const setStyle = ( newValue ) => {
+		const newUserConfig = cloneDeep( userConfig );
+		set( newUserConfig, finalPath, newValue );
+		setUserConfig( newUserConfig );
+	};
+
+	let result;
+	switch ( source ) {
+		case 'all':
+			result =
+				get( userConfig, finalPath ) ?? get( baseConfig, finalPath );
+			break;
+		case 'user':
+			result = get( userConfig, finalPath );
+			break;
+		case 'base':
+			result = get( baseConfig, finalPath );
+			break;
+		default:
+			throw 'Unsupported source';
+	}
+
+	return [ result, setStyle ];
+}
+
+const ROOT_BLOCK_SUPPORTS = [
+	'background',
+	'backgroundColor',
+	'color',
+	'linkColor',
+	'fontFamily',
+	'fontSize',
+	'fontStyle',
+	'fontWeight',
+	'lineHeight',
+	'textDecoration',
+	'textTransform',
+	'padding',
+];
+
+export function getSupportedGlobalStylesPanels( name ) {
+	if ( ! name ) {
+		return ROOT_BLOCK_SUPPORTS;
+	}
+
+	const blockType = getBlockType( name );
+
+	if ( ! blockType ) {
+		return [];
+	}
+
+	const supportKeys = [];
+	Object.keys( STYLE_PROPERTY ).forEach( ( styleName ) => {
+		if ( ! STYLE_PROPERTY[ styleName ].support ) {
+			return;
+		}
+
+		// Opting out means that, for certain support keys like background color,
+		// blocks have to explicitly set the support value false. If the key is
+		// unset, we still enable it.
+		if ( STYLE_PROPERTY[ name ].requiresOptOut ) {
+			if (
+				has(
+					blockType.supports,
+					STYLE_PROPERTY[ name ].support[ 0 ]
+				) &&
+				get( blockType.supports, STYLE_PROPERTY[ name ].support ) !==
+					false
+			) {
+				return supportKeys.push( name );
+			}
+		}
+
+		if (
+			get( blockType.supports, STYLE_PROPERTY[ name ].support, false )
+		) {
+			return supportKeys.push( name );
+		}
+	} );
+
+	return supportKeys;
+}
```

Comment:
The new hooks useSetting and useStyle might be a bit less performant than the previous APIs in. `global-styles-provider` (which we should try to remove) because I'm not using "context" at all. I believe it's something we can improve later if we do notice issues, right now I wanted to focus on DevX